src/main/java/life/qbic/omero/BasicOMEROClient.java

```diff
@@ -531,13 +568,29 @@ public void addMapAnnotationToDataset(long datasetId, String key, String value)
 
 
 
-    } catch (Exception e) {
-      System.out.println(e);
+    } catch (DSOutOfServiceException dsOutOfServiceException) {
+      throw new RuntimeException("Error while accessing omero service: broken connection, expired session or not logged in", dsOutOfServiceException);
+    } catch (ExecutionException executionException) {
+      throw new RuntimeException("Task aborted unexpectedly.", executionException);
+    } catch (DSAccessException dsAccessException) {
+      throw new RuntimeException("Could not pull data from the omero server.", dsAccessException);
     }
 
     return imageInfo;
   }
 
+  public String imageViewerLink(long imageId) {
```

Comment:
Not a good method name, a comment would also be helpful

Reply:
What would you suggest for a method name? I do not want to create a fake getter method.

Reply:
good point.
[create|return|fetch]ImageViewerLink?

Reply:
create -> no constructor
return -> dangerous in Java
fetch -> not really fetching anything

Reply:
createImageViewLink() ?

Reply:
If we do it via a file then what about parse?

Reply:
> If we do it via a file then what about parse?

We don't do it over a file. We just concatenate instance properties here with a little mumbo jumbo.

Reply:
build?

Reply:
I do like `omeroClient.imageViewerLink(imageid)`. It is very descriptive in my opinion.

Reply:
I guess there are worse names. ;)
Then let's keep it, but maybe we should discuss these conventions in the future. Functions without verb feel strange to me, but I might be wrong.

Please add the Java Doc header, though.

Reply:
what about `linkImageViewer()`?

Reply:
That sounds like the function links the Image Viewer...maybe it does, though?

Reply:
Or `queryImageViewerAddress`. I know it does not query anything strictly speaking but the caller queries the client for the address.

Reply:
> Method names are written in lowerCamelCase.
Method names are typically verbs or verb phrases. For example, sendMessage or stop.Underscores may appear in JUnit test method names to separate logical components of the name, with each component written in lowerCamelCase. One typical pattern is <methodUnderTest>_<state>, for example pop_emptyStack. There is no One Correct Way to name test methods.


Source: https://google.github.io/styleguide/javaguide.html#s5.1-identifier-names

Reply:
That is why I changed my mind and suggested methods with verbs :) Found the same

Reply:
In general I would try to be as descriptive as possible with the method name:
`retrieveOmeroLinkForId()`
`generateOmeroLinkForId()`
...
something like that.

Reply:
I chose to go with `composeImageDetailAddress` since it basically composes different information into a link to the image detail view provided by the omero web client.